src/lib.rs

```diff
@@ -270,4 +272,75 @@ where
         let result: i16 = (bytes[1] as i16) << 8 | bytes[0] as i16;
         Ok((result as f32) / TEMP_SCALE + TEMP_BIAS)
     }
+
+    /// Enable and configure FIFO
+    pub fn configure_fifo(&mut self, config: FIFOConfig) -> Result<(), T::Error> {
+        // write values to the FIFO_CTRL register
+        self.interface.write(
+            Sensor::Accelerometer,
+            register::AG::FIFO_CTRL.addr(),
+            config.f_fifo_ctrl(),
+        )?;
+
+        // write values to specific bits of the CTRL_REG9 register
+        let ctrl_reg9: u8 =
+            self.read_register(Sensor::Accelerometer, register::AG::CTRL_REG9.addr())?;
+        let data: u8 = config.f_ctrl_reg9();
+        let mut payload: u8 = ctrl_reg9 & !FIFOBitmasks::CTRL_REG9_FIFO;
+        payload |= data;
+        self.interface.write(
+            Sensor::Accelerometer,
+            register::AG::CTRL_REG9.addr(),
+            payload,
+        )?;
+
+        Ok(())
+    }
+
+    /// Get flags and FIFO level from the FIFO_STATUS register
+    pub fn get_fifo_status(&mut self) -> Result<FIFOStatus, T::Error> {
+        let fifo_src = self.read_register(Sensor::Accelerometer, register::AG::FIFO_SRC.addr())?;
+        let fifo_level_value = fifo_src & FIFOBitmasks::FSS;
+        let status = FIFOStatus {
+            /// Is FIFO filling equal or higher than the threshold?
+            fifo_thresh_reached: match fifo_src & FIFOBitmasks::FTH {
+                0 => false,
+                _ => true,
+            },
+            /// Is FIFO full and at least one sample has been overwritten?
+            fifo_overrun: match fifo_src & FIFOBitmasks::OVRN {
+                0 => false,
+                _ => true,
+            },
+            /// Is FIFO empty (no unread samples)?
+            fifo_empty: match fifo_level_value {
+                0 => true,
+                _ => false,
+            },
```

Comment:
Do we want to use `match` here? Can we just use boolean expressions instead?

Reply:
OK, will change to boolean expressions as per one of your emails. 

Should I just change it and commit again? Will that work?

Reply:
Yes, please make the changes and push.

Reply:
OK, done.